server.js

```diff
@@ -1,29 +1,125 @@
 import express from 'express'
-import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import dotenv from 'dotenv'
+import listEndpoints from 'express-list-endpoints'
 
+dotenv.config()
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+mongoose.connect(mongoUrl, { 
+  useNewUrlParser: true, 
+  useUnifiedTopology: true, 
+  useCreateIndex: true 
+})
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
+const port = process.env.PORT || 8082
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+const thoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  }
+})
+
+const Thought = mongoose.model('Thought', thoughtSchema)
+
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  try {
+    res.send(listEndpoints(app))
+  } catch (error) {
+    res.status(404).json({ error: 'Page not found' })
+  }
+})
+
+app.get('/thoughts', async (req, res) => {
+  try {
+    const allThoughts = await Thought.find().sort({ createdAt: -1 }).limit(20)
+    res.json(allThoughts)
+  } catch (error) {
+    res.status(404).json({ error: 'Page not found' })
+  }
+})
+
+app.post('/thoughts', async (req, res) => {
+  try {
+    const newThought = await new Thought({
+      message: req.body.message
+    }).save()
+    res.json(newThought)  
+  } catch (error) {
+    if (error.code === 11000) {
+      res.status(400).json({ error: "Duplicated value", fields: error.keyValue })
+    }
+    res.status(400).json(error)
+  }
+})
+app.post('/thoughts', async (req, res) => {
+  try {
+    const newThought = await new Thought({
+      message: req.body.message
+    }).save()
+    res.json(newThought)  
+  } catch (error) {
+    if (error.code === 11000) {
+      res.status(400).json({ error: "Duplicated value", fields: error.keyValue })
+    }
+    res.status(400).json(error)
+  }
+})
+
+app.delete('/thoughts/delete/:id', async (req, res) => {
+  const { id } = req.params
+  try {
+    const updatedThoughts = await Thought.findByIdAndDelete({
+      _id: id
+    })
+    res.json(updatedThoughts)
+  } catch (error) {
+    res.status(400).json({ error: "Could not delete", fields: error.keyValue })
+  }
+})
```

Comment:
Remember to restrict access to this endpoint because not all users should have access to this functionality 🤓